core/src/main/scala/org/apache/spark/rpc/netty/NettyRpcEnv.scala

```diff
@@ -451,7 +451,6 @@ private[netty] class NettyRpcEnv(
     }
 
     override def onFailure(streamId: String, cause: Throwable): Unit = {
-      logDebug(s"Error downloading stream $streamId.", cause)
```

Comment:
So, the deadlock happens when we use DEBUG log level?

Reply:
the log4j will loadClass again with `ExecutorClassLoader`, if the loglevel does not satisfy, it is skipped

Reply:
Before the deadlock occurs, I see something very similar with https://github.com/apache/spark/pull/10337#issuecomment-165266841